File: packages/shared/src/types/usage.ts
export interface WorkspaceUsageSummary {
  period: { from: string; to: string };
  totalRuns: number;
  successfulRuns: number;
  failedRuns: number;
  totalTokens: { input: number; output: number } | null;
  estimatedCost: number | null;
  agentBreakdown: AgentUsage[];
  modelBreakdown: ModelUsage[];
  dailyTrend: DailyUsage[];
  dataAvailable: boolean;
}

export interface AgentUsage {
  agentId: string;
  agentName: string;
  runCount: number;
  successRate: number;
  totalTokens: { input: number; output: number } | null;
  estimatedCost: number | null;
}

export interface ModelUsage {
  model: string;
  runCount: number;
  totalTokens: { input: number; output: number } | null;
  estimatedCost: number | null;
}

export interface DailyUsage {
  date: string;
  runCount: number;
  cost: number | null;
}

File: packages/ios/MulticaConsole/Models/Usage.swift
import Foundation

struct UsageSummary: Codable {
    struct Period: Codable {
        let from: String
        let to: String
    }

    let period: Period
    let totalRuns: Int
    let successfulRuns: Int
    let failedRuns: Int
    let totalTokens: TokenUsage?
    let estimatedCost: Double?
    let agentBreakdown: [TopAgent]
    let dailyTrend: [DailyUsage]?
    let dataAvailable: Bool
}

struct TopAgent: Codable, Identifiable {
    var id: String { agentId }
    let agentId: String
    let agentName: String
    let runCount: Int
    let successRate: Double
    let totalTokens: TokenUsage?
    let estimatedCost: Double?
}

struct DailyUsage: Codable {
    let date: String
    let runCount: Int
    let cost: Double
}
